bats/plan.sh

```diff
@@ -28,4 +29,21 @@ do_check() {
 
 do_install() {
   ./install.sh "$pkg_prefix"
+
+  _install_library "bats-mock" "https://github.com/lox/bats-mock/archive/master.tar.gz" "stub.bash"
```

Comment:
I wonder if these libraries are better packaged as separate plans. As these update and are changed, releasing a new version of them will be difficult to track, as its embedded in the bats plan.

How do you feel about creating separate plans for these, and dependencies in bats on those new plans? @tduffield

Reply:
So that was my initial plan, but I thought about our pattern for other things like gems and nodejs, where the pattern was not to recreate a hab package for every addon and library available. 

I went back and forth on whether or not to even include these libraries in the core package at all, because of that pattern. I eventually decided to include them because these libraries honestly don't change much. Bats itself has been at 0.4.0 since 2014. bats-support and bats-support haven't been updated since 2016. The upstream of the bats-mock I included had been quiet since 2015, and was only recently updated in the fork i provided in Oct 2018 to fix some minor issues. 

I guess my point is that I don't think there is a ton of value in individual packages for each library, I doubt that there will be many (if any) substantial changes to these libraries in the future, and there is a lot of value in having these libraries all prepped and ready to go.

Reply:
I find having these libraries in the `core/bats` plan useful. If we don't want to have these libraries in the `core/bats` plan, I honestly think it'd be better to not include them in core at all. 

It that's the way the rest of the maintainers feel, I'm happy to withdraw this PR. Thought, to be honest, I'll just publish this plan in my own origin, since I find having these libs already packages with my bats SUPER useful.

Reply:
I've got mixed feelings on this and have been going back and forth all morning.  In general, I think core should "follow upstream" and the package should be as vanilla as possible,  but that should be a guideline and not a hard rule. 

Libraries are a difficult thing with Habitat currently.  I agree with @tduffield that they (probably) don't belong in core as individual packages. I also think that _in general_ bundling them (in core at least) isn't the right pattern either.

Versioning of libraries or embedded software is a thing that has bitten us in the past (see [rubygems](https://github.com/habitat-sh/core-plans/pull/2235) in the last refresh).  As pointed out though, these libraries don't change much so the risk is minimal. 

Adding libraries to a package can also bloat the dependencies and installed size, postgis in the postgresql package being an extreme example that almost doubles the installed sized and number of dependencies due to gdal and proj. That's not the case with this change however as it's bash all the way down. 

I'm leaning towards accepting this, it adds value to the package, the risk is minimal(from versioning and bloat), and the libraries are opt-in at runtime. 

I just would like to make sure we make it clear to our future selves and maintainers why we'd accept this change, but reject a similar change on other packages. 

I'm going give @predominant a chance to respond when he comes online, and barring objections will hit the button (or he can 😄 )

Reply:
Ain't no-one hitting that button until DCO check is completed successfully!

I'm on the fence on this one.

I don't want this to become common practice, nor for this to be a precedent on which similar merges are done in the past. Including additional libraries that may or may not be used, and that are easy added in other packages seems like a good reason to separate them into other packages. Minimise bloat, and maintain core-plans are a pristine example of how to package and separate concerns with plans.

On the other side of the fence, they are small libraries, so the impact is minimal.

That being said, I know this will become a problem when an embedded library version is changed.

I'm leaning towards not accepting this. But would accept in an instant if they were separate plans.

Reply:
What is they were all together in a separate plan that depended on `core/bats`? We could call that plan `core/bats-plus` or `core/bats-libs`. That seems like a plausible middle ground, and gives us a good pattern. 

For less active libraries like these, maybe they all exist in a single package. 
For more active libraries, they need to exist as individual packages, and /could/ be wrapped up into a single package? 

Also, I'm not sure why DCO hasn't been reported, I DCO sign every commit I make to everything.